ftcosi/protocol/protocol.go

```diff
@@ -127,72 +129,88 @@ func (p *FtCosi) Dispatch() error {
 
 	// generate trees
 	nNodes := p.Tree().Size()
-	trees, err := genTrees(p.Tree().Roster, nNodes, p.NSubtrees)
+	trees, err := genTrees(p.Tree().Roster, p.Tree().Root.RosterIndex, nNodes, p.NSubtrees)
 	if err != nil {
+		p.FinalSignature <- nil
 		return fmt.Errorf("error in tree generation: %s", err)
 	}
 
-	// if one node, sign without subprotocols
-	if nNodes == 1 {
+	// if one node or threshold is one, sign without subprotocols
+	if nNodes == 1 || p.Threshold == 1 {
 		trees = make([]*onet.Tree, 0)
 	}
 
 	// start all subprotocols
-	cosiSubProtocols := make([]*SubFtCosi, len(trees))
+	p.subProtocols = make([]*SubFtCosi, len(trees))
 	for i, tree := range trees {
-		cosiSubProtocols[i], err = p.startSubProtocol(tree)
+		p.subProtocols[i], err = p.startSubProtocol(tree)
 		if err != nil {
+			p.FinalSignature <- nil
 			return err
 		}
 	}
 	log.Lvl3(p.ServerIdentity().Address, "all protocols started")
 
-	commitments, runningSubProtocols, err := p.collectCommitments(trees, cosiSubProtocols)
+	// collect commitments
+	commitments, runningSubProtocols, err := p.collectCommitments(trees, p.subProtocols)
 	if err != nil {
+		p.FinalSignature <- nil
 		return err
 	}
 
-	// signs the proposal
-	ok := <-verifyChan
-	if !ok {
-		// root should not fail the verification otherwise it would not have
-		// started the protocol
+	var secret kyber.Scalar
+	// verifies the proposal
+	verificationOk := <-verifyChan
```

Comment:
might be a good idea to add a timeout, just in case the verification function doesn't stop

Reply:
added timeout while waiting for the verification.